### PR TITLE
Wallpaper module 🙄

### PR DESCRIPTION
**Pull request by NobArxtx**
Idk what to write here

### CODE REVIEW

**Comment by Phyco-Ninja** [comment]
Omg wow 😐 current record holder is u

**Comment by rowdy-hero** [comment]
He breaks my record too.. 😂😂

**Comment by Muttahir786** [comment]
tabs oh oh

**Comment by Phyco-Ninja** [comment]
> He breaks my record too..

👀 Soon I willl regain my record with new modules and make it longer than this;) just wait sar

**Comment by cyberboysumanjay** [comment]
Welcome to fuck pep8 club 🥳

**Comment by Phyco-Ninja** [comment]
> Welcome to fuck pep8 club

Omg take birth control pills then 🌚🤧

**Comment by Phyco-Ninja** [comment]
WeW, deepsource too have issues 🤔

**Comment by NobArxtx** [comment]
Oof 😶

**Comment by prono69** [comment]
@cyberboysumanjay 

Hehe Boi :new_moon_with_face:

_Sent from my Redmi Note 4 using [FastHub-Libre](https://github.com/thermatk/FastHub-Libre/)_

**Comment by Phyco-Ninja** [comment]
> @cyberboysumanjay
> 
> Hehe Boi 
> 
> _Sent from my Redmi Note 4 using [FastHub-Libre](https://github.com/thermatk/FastHub-Libre/)_

Don't advertise sar 🌚 we know u is reech

**Comment by prono69** [comment]
@Phyco-Ninja sar it's default sar .. idk how to remove dat ..

Me not Rich .. u r :) ❤️

_Sent from my Redmi Note 4 using [FastHub-Libre](https://github.com/thermatk/FastHub-Libre/)_

**Comment by Phyco-Ninja** [comment]
> @Phyco-Ninja sar it's default sar .. idk how to remove dat ..
> 
> Me not Rich .. u r :) 
> 
> _Sent from my Redmi Note 4 using [FastHub-Libre](https://github.com/thermatk/FastHub-Libre/)_

Hehe, not disclosing wealth in public 🌚 peru moves sar;)

**Comment by rking32** [comment]
@NobArxtx this plugin worked for you ?

**Comment by Harsh-br0** [comment]
Omg
Wth is going on

**Comment by Phyco-Ninja** [comment]
> @Phyco-Ninja
> Now u proving u r a real spammer

What do u mean by proving sar 🌚🤧 
Me not spammer sar.

**Comment by Harsh-br0** [comment]
@Phynco-ninja btw where is ur pep8 leaderboards 😎

**Comment by Mongodbuser-red** [comment]
> @NobArxtx this plugin worked for you ?

Yeap

**Comment by NobArxtx** [comment]
> @NobArxtx this plugin worked for you ?

Yeap

**Comment by NobArxtx** [comment]
> @NobArxtx this plugin worked for you ?

Yeap

File: plugins/wall.py [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        for i in wall_num:
+            try:
+                wall_num = int(i)
+            except ValueError:
+                pass
+        try:
+            page_num = resp.find('div', {'class': 'visible-xs'})
+            page_num = page_num.find('input', {'class': 'form-control'})
+            page_num = int(page_num['placeholder'].split(' ')[-1])
+        except Exception as e:
```
Comment (rking32) [inline plugins/wall.py:45]:
```suggestion
        except Exception:
```

File: plugins/wall.py [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+            page_num = 1
+
```
Comment (rking32) [inline plugins/wall.py:47]:
```suggestion

```

File: plugins/wall.py [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            await message.edit(e)
+            return
+        if link:
+            await message.edit('**Processing...**')
+            idl = await dlimg(link[0])
+            await message.edit('**Uploading...**')
+            if not len(link[1].split()) < 11:
+                capo = '**' + ' '.join(link[1].split()[:11]) + '**'
+            else:
+                capo = '**' +link[1]+ '**'
```
Comment (rking32) [inline plugins/wall.py:110]:
```suggestion
                capo = '**' + link[1] + '**'
```

**Comment by rking32** [comment]
@NobArxtx can you commit these changes

**Comment by rking32** [comment]
> Change it to point_down
> with open.(path_i, 'wb') as k:
> k.write(e)
> k.close

```python
with open.(path_i, 'wb') as k:
     k.write(e)
```

**Comment by Phyco-Ninja** [comment]
Still a typo in line 13  🤔

**Comment by rowdy-hero** [comment]
Finally :-|

File: plugins/wall.py [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+import os
+
+
+async def dlimg(link):
+    e = requests.get(link).content
+    paea = 'donno.{}'.format(link.split('.')[-1])
+    path_i = os.path.join(Config.DOWN_PATH, paea)
+    with open(path_i, 'wb') as k:
+        k.write(e)
+        k.close()
```
Comment (rking32) [inline plugins/wall.py:14]:
```suggestion
        k.write(e)
```

**Comment by NobArxtx** [comment]
GitHub file editor bugs 😩

**Comment by Phyco-Ninja** [comment]
> k = open(path_i, 'wb')
> k.write(e)
> k.close()
> 
> Change it to 
> with open.(path_i, 'wb') as k:
> k.write(e)
> k.close
> 
> 
> 
> Line 119:
> use os.remove(idl)
> 
> Remove line 32 and 49 and fix indents

@NobArxtx Check this also plox :)

**Comment by Phyco-Ninja** [comment]
#MakeDeepSourceHappy

File: plugins/wall.py [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+            except ValueError:
+                pass
+        try:
+            page_num = resp.find('div', {'class': 'visible-xs'})
+            page_num = page_num.find('input', {'class': 'form-control'})
+            page_num = int(page_num['placeholder'].split(' ')[-1])
+        except Exception:
+            page_num = 1
+        n = randint(1, page_num)
+        if True:
```
Comment (rking32) [inline plugins/wall.py:48]: what is this ?
True always True

File: plugins/wall.py [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+    page_link = 'https://wall.alphacoders.com/search.php?search={}&page={}'
+    resp = requests.get(f'{url}{strin}')
+    if resp.url == none_got:
+        return False
+    if 'by_category.php' in resp.url:
+        page_link = str(resp.url).replace('&amp;', '') + '&page={}'
+        check_link = True
+    else:
+        check_link = False
+    if True:
```
Comment (rking32) [inline plugins/wall.py:32]: this also

File: plugins/wall.py [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+                capo = '**' + link[1] + '**'
+            try:
+                cat_id = message.chat.id
+                await userge.send_photo(cat_id, idl, caption=capo)
+                await userge.send_document(message.chat.id, idl)
+            except Exception as e:
+                await message.edit(e)
+                return
+            await message.delete()
+            os.system('rm {}'.format(idl))
```
Comment (rking32) [inline plugins/wall.py:118]: don't use shell directly
try subprocess

Comment (Phyco-Ninja) [reply plugins/wall.py:118]: 🌚 we habe  os.remove

Comment (NobArxtx) [reply plugins/wall.py:118]: Yeap 
I will try

Comment (Harsh-br0) [reply plugins/wall.py]: > Yeap
> I will try

Still u using requests
@SpEcHiDe suggested u to use aiohttp🤔

**Comment by Phyco-Ninja** [comment]
Finally u used os.remove 🌚🤧

**Comment by rking32** [comment]
@NobArxtx now looks cool, but one thing. you doing this in hard way

https://github.com/UsergeTeam/Userge-Plugins/blob/660fe03fbbffa2b1f08867fc649c6c3cea54c00b/plugins/wall.py#L90-L93

simply use like this

add this on top
```python
CHANNEL = userge.getCLogger(__name__)
```

and

```python
await CHANNEL.log(f"Search Query : {message.input_str}")
```

**Comment by rking32** [comment]
@NobArxtx voila ... :beers: , merging ...

**Pull request merged**